src/Analyser/NodeScopeResolver.php

```diff
@@ -951,7 +951,10 @@ private function processStmtNode(
 						]),
 					)));
 				} else {
-					$finalScope = $finalScope->mergeWith($scope);
+					$finalScope = $finalScope->mergeWith($scope->filterByTruthyValue(new BinaryOp\Identical(
+						$stmt->expr,
+						new Array_([]),
+					)));
```

Comment:
Please remove the if-else as both these branches look like should be the same.

Reply:
done